core/arch/arm/kernel/abort.c

```diff
@@ -32,23 +32,58 @@
 #include <kernel/user_ta.h>
 #include <kernel/unwind.h>
 #include <mm/core_mmu.h>
+#include <mm/mobj.h>
 #include <mm/tee_pager.h>
 #include <tee/tee_svc.h>
 #include <trace.h>
 #include <arm.h>
 
+#include "thread_private.h"
+
 enum fault_type {
 	FAULT_TYPE_USER_TA_PANIC,
 	FAULT_TYPE_USER_TA_VFP,
 	FAULT_TYPE_PAGEABLE,
 	FAULT_TYPE_IGNORE,
 };
 
-#ifdef CFG_CORE_UNWIND
+#ifdef CFG_UNWIND
 #ifdef ARM32
-static void __print_stack_unwind(struct abort_info *ai)
+
+/*
+ * These are set in the linker script. Their addresses will be the start or end
+ * of the exception binary search index table (.ARM.exidx section)
+ */
+extern uint8_t __exidx_start[];
+extern uint8_t __exidx_end[];
+
+/*
+ * Kernel or user mode unwind (32-bit execution state).
+ */
+static void __print_stack_unwind_arm32(struct abort_info *ai)
 {
-	struct unwind_state state;
+	struct unwind_state_arm32 state;
+	uaddr_t exidx;
+	size_t exidx_sz;
+
+	EMSG_RAW("Call stack:");
+	if (abort_is_user_exception(ai)) {
+		struct tee_ta_session *s;
```

Comment:
What's in this block could go into a separate function that could be called from the other version of `__print_stack_unwind_arm32()` below.